src/psyclone/tests/psyGen_test.py

```diff
@@ -3792,6 +3908,245 @@ def test_fparser2astprocessor_handling_if_stmt(f2008_parser):
     assert len(new_node.children) == 2
 
 
+def test_fparser2astprocessor_handling_if_construct(f2008_parser):
+    ''' Test that fparser2 If_Construct is converted to the expected PSyIR
+    tree structure.
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Execution_Part
+    reader = FortranStringReader(
+        '''if (condition1 == 1) then
+            branch1 = 1
+        elseif (condition2 == 2) then
+            branch2 = 1
+        else
+            branch3 = 1
+        endif''')
+    fparser2if_construct = Execution_Part.match(reader)[0][0]
+
+    fake_parent = Node()
+    processor = Fparser2ASTProcessor()
+    processor.process_nodes(fake_parent, [fparser2if_construct], None)
+
+    # Check a new node was properly generated and connected to parent
+    assert len(fake_parent.children) == 1
```

Comment:
Could you add checks that the various `._ast` and `._ast_end` are set correctly here too?

Reply:
Added some `_ast/_ast_end` checks on `test_fparser2astprocessor_handling_if_construct`

Reply:
Also added checks in `test_fparser2astprocessor_handling_Case_construct`

Reply:
Great, thanks.